usr/local/www/diag_ping.php

```diff
@@ -79,118 +89,102 @@
-		$ifscope = '';
-		$command = "/sbin/ping";
-		if ($ipproto == "ipv6") {
-			$command .= "6";
-			$ifaddr = is_ipaddr($sourceip) ? $sourceip : get_interface_ipv6($sourceip);
-			if (is_linklocal($ifaddr))
-				$ifscope = get_ll_scope($ifaddr);
-		} else {
-			$ifaddr = is_ipaddr($sourceip) ? $sourceip : get_interface_ip($sourceip);
-		}
-		if ($ifaddr && (is_ipaddr($host) || is_hostname($host))) {
-			$srcip = "-S" . escapeshellarg($ifaddr);
-			if (is_linklocal($host) && !strstr($host, "%") && !empty($ifscope))
-				$host .= "%{$ifscope}";
-		}
-
-		$cmd = "{$command} {$srcip} -c" . escapeshellarg($count) . " " . escapeshellarg($host);
-		//echo "Ping command: {$cmd}\n";
-		system($cmd);
-		echo('</textarea>&nbsp;</font>');
+	$ifscope = '';
+	$command = "/sbin/ping";
+	if ($ipproto == "ipv6") {
+		$command .= "6";
+		$ifaddr = is_ipaddr($sourceip) ? $sourceip : get_interface_ipv6($sourceip);
+		if (is_linklocal($ifaddr))
+			$ifscope = get_ll_scope($ifaddr);
+	} else {
+		$ifaddr = is_ipaddr($sourceip) ? $sourceip : get_interface_ip($sourceip);
 	}
-	?>
-	</td>
-</tr>
-<tr>
-	<td width="22%" valign="top">&nbsp;</td>
-	<td width="78%">&nbsp;</td>
-</tr>
-</table>
-</form>
-</td></tr>
-</table>
-<?php include("fend.inc"); ?>
-</body>
-</html>
+
+	if ($ifaddr && (is_ipaddr($host) || is_hostname($host))) {
+		$srcip = "-S" . escapeshellarg($ifaddr);
+		if (is_linklocal($host) && !strstr($host, "%") && !empty($ifscope))
+			$host .= "%{$ifscope}";
+	}
+
+	$cmd = "{$command} {$srcip} -c" . escapeshellarg($count) . " " . escapeshellarg($host);
+	//echo "Ping command: {$cmd}\n";
+	$result = shell_exec($cmd);
+
+	if(empty($result))
+		$input_errors[] = "Host \"" . $host . "\" did not respond or could not be resolved.";
+
+}
+
+include("head.inc"); ?>
+
+<?php if ($input_errors)
+	print_input_errors($input_errors);
+
+require('classes/Form.class.php');
+
+$form = new Form(new Form_Button(
+	'Submit',
+	'Ping'
+));
+
+$section = new Form_Section('Traceroute');
+
+$section->addInput(new Form_Input(
+	'host',
+	'Hostname',
+	'text',
+	$host,
+	['placeholder' => 'Hostname to ping.']
+));
+
+$section->addInput(new Form_Select(
+	'ipproto',
+	'IP Protocol',
+	$pconfig['protocol'],
+	array('ipv4' => 'IPv4', 'ipv6' => 'IPv6')
+))->setHelp('Select the protocol to use');
+
+$section->addInput(new Form_Select(
+	'sourceip',
+	'Source Address',
+	$pconfig['source'],
+	create_sourceaddresslist()
+))->setHelp('Select source address for the ping');
+
+$section->addInput(new Form_Select(
+	'count',
+	'Maximum nuber of pings',
+	$count,
+	array_combine(range(1, MAX_COUNT), range(1, MAX_COUNT))
+))->setHelp('Select the maximum number pings');
+
+$form->add($section);
+print $form;
+
+if($do_ping && !empty($result) && !$input_errors) {
+	print("<div class=\"panel panel-info\">" . "<div class=\"panel-heading\">Results</div>");
+	print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));
+	print('</div></div>');
+}
+
+include("foot.inc");
+
```

Comment:
Updated per your suggestions, except that I would like to leave:

    	// ToDo: Revert to nl2br() and remove &nbsp;when the CSS has been adjusted
	print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));

in place for now. I am asked to demonstrate these pages from time time and I need them to look good even without the planned CSS adjustment :)

Steve


> On Apr 14, 2015, at 1:53 PM, SjonHortensius <notifications@github.com> wrote:
> 
> In usr/local/www/diag_ping.php <https://github.com/SjonHortensius/pfsense/pull/69#discussion_r28351721>:
> 
> > +	'Maximum nuber of pings',
> > +	$count,
> > +	array_combine(range(1, MAX_COUNT), range(1, MAX_COUNT))
> > +))->setHelp('Select the maximum number pings');
> > +
> > +$form->add($section);
> > +print $form;
> > +
> > +if($do_ping && !empty($result) && !$input_errors) {
> > +	print("<div class=\"panel panel-info\">" . "<div class=\"panel-heading\">Results</div>");
> > +	print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));
> > +	print('</div></div>');
> > +}
> > +
> > +include("foot.inc");
> > +
> Please delete the last 3 lines. We don't need the closing tag and whitespace is unneeded as well (including the last enter)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/SjonHortensius/pfsense/pull/69/files#r28351721>.
> 

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com <mailto:sbeaver@netgate.com> | www.netgate.com <http://www.netgate.com/>